nrf_rpc/doc/architecture.rst

```diff
@@ -0,0 +1,96 @@
+.. architecture:
+
+
+Architecture
+############
+
+The following picture gives an overview of the nRF_RPC architecture:
+
+.. image:: img/layers.svg
+   :alt: nRF_RPC Architecture Layers
+   :align: center
+
+
+Types of communication
+======================
+
+Calls can be done in one of two ways: using a command/response or an event.
+
+**Command and response** are intended to be used for a synchronous function calls.
+The caller sends the command to the other side and waits for a response.
+The response is send after end of execution of the a remote function.
+Following image shows an example of the command and response flow.
+
+.. image:: img/cmd_flow.svg
+   :alt: nRF_RPC Command/Response flow
+   :align: center
+
+**Events** are intended to be used for an asynchronous function calls.
+The caller sends an event to the other side and returns immediately if there is free thread in thread pool, otherwise it waits for available thread.
```

Comment:
The concept of the Thread pool needs to be introduced before this.  Maybe up in the architecture section it could briefly introduce the idea that the remote must have a pool of one or more Threads - a Thread is used for each call.

Does this mean that every call made through RPC must be re-entrant and the APIs all thread safe or is there a way to restrict which calls can be made concurrently?

Reply:
nRF RPC API can be called from many threads without any restrictions. Only limitation is the thread pool size on the remote side. If it is fully used then call must wait (block) until some remote thread will be available.

Reply:
Maybe adding `(see :ref:'Threads' section)` is enough?

> The caller sends an event to the other side and returns immediately if there is free thread in thread pool **(see [Threads](#) section)**, otherwise it waits for available thread.